### PR TITLE
Rationalize dependencies sources

### PR DESCRIPTION
**Pull request by A669015**
The goal is to centralize the reference list of python dependencies in a single file: the `requirements.txt`.

The `torch` version is not in the `requirements.txt` to keep the philosopy of a base env (docker or conda) providing `python `and `torch`: 
 - conda: only one yaml file is kept (`env.yaml`), that is used to reference the `python` and `torch` versions, and install the python dependencies with pip from the  `requirements.txt`,
 - docker: kept as it is (`torch` version provided by the base pytorch image, but **warning** with the `python` version that is not mastered in this case), just a bit of parametrization of `cuda` and `torch` as been added.

### CODE REVIEW

File: env.yaml
Comment (cbovalo) [inline env.yaml:1]: Maybe we could even upgrade torch to version 2.3 ?

Comment (A669015) [reply env.yaml:1]: I would prefer manage the upgrade  in a futher PR.

**Review by cbovalo — APPROVED** [review]

**Comment by flyIchtus** [comment]
Hey, can we move onnxruntime dependencies from ==1.17.xxx to >=1.16 ? this would allow to make the install work on MF's HPC.

**Comment by colon3ltocard** [comment]
@A669015 @cbovalo : if you use docker on your machine could you add to the README.md the full commands to build the image and run a training ? I think it will be valuable for others.

File: requirements.txt
[the remark this answers is not in the document]
Comment (A669015) [reply requirements.txt:1]: `python bin/train.py --dataset titan --model hilam --dataset_conf config/datasets/titan_full.json --dev_mode --no_log --num_pred_steps_val_test 1 --num_input_steps 1` succeeded with only `torch-geometric` installed (none of its dependencies were installed), so I guess we can remove them

**Review by cbovalo — APPROVED** [review]

**Review by colon3ltocard — APPROVED** [review]

**Comment by colon3ltocard** [comment]
> Hey, can we move onnxruntime dependencies from ==1.17.xxx to >=1.16 ? this would allow to make the install work on MF's HPC.

I merged @A669015, let's make a dedicated PR if we still need some fixes for running on Belenos.

**Pull request merged**